### PR TITLE
RSDK-1729 - change resolution constraints to be of type IntExact

### PR DESCRIPTION
**Pull request by Rob1in**
change resolutions constraints to be IntExact instead of IntRanged.

### CODE REVIEW

File: components/camera/videosource/webcam.go [hunk 1]
```diff
@@ -147,29 +147,19 @@ type WebcamAttrs struct {
...
-	}
-
 	return mediadevices.MediaStreamConstraints{
 		Video: func(constraint *mediadevices.MediaTrackConstraints) {
-			constraint.Width = prop.IntRanged{minWidth, maxWidth, idealWidth}
-			constraint.Height = prop.IntRanged{minHeight, maxHeight, idealHeight}
+			if attrs.Width > 0 {
+				constraint.Width = prop.IntExact(attrs.Width)
+			} else {
+				constraint.Width = prop.IntRanged{Min: 0, Ideal: 640, Max: 4096}
```
Comment (bazile-clyde) [inline components/camera/videosource/webcam.go:155]: Why did the ideal width change from 800 to 640? Same question for height changing from 600 to 480.

Comment (Rob1in) [reply components/camera/videosource/webcam.go:155]: This is more aligned with the "default driver" (first) proposed by v4l for most webcams. Also, in the fitnessDistance algorithm of mediadevices, only min and max will matter.

Comment (bazile-clyde) [reply components/camera/videosource/webcam.go:155]: Mediadevices uses `Ideal` too in its fitnessDistance algorithm. It uses it when comparing one `IntRanged` to another (see [IntRanged.Compare](https://github.com/pion/mediadevices/blob/f8f8511d94f68cf126c49b85bda29e3f558b0735/pkg/prop/int.go#L86-L116)). The `fitnessDistance` method will compare `IntRanged`s (a type of `IntConstraint`) [here](https://github.com/pion/mediadevices/blob/f8f8511d94f68cf126c49b85bda29e3f558b0735/pkg/prop/prop.go#L185).

All that being said I don't think it matters too much. This looks LGTM!

**Review by bazile-clyde — APPROVED** [review]

**Review by bhaney — APPROVED** [review]
Looks good to me, making so it matches exactly to the specified resolution will finally fix the matching bug we've seen

**Pull request merged**